### PR TITLE
add kotlinx-io library

### PR DESCRIPTION
**Pull request by btwonion**
This PR adds the [kotlinx-io](https://github.com/Kotlin/kotlinx-io) library.
It is used by some mods and fits into the libraries this mod already provides.

### CODE REVIEW

**Comment by modmuss50** [comment]
Can you add this to the readme? See the template here: https://github.com/FabricMC/fabric-language-kotlin/blob/master/templates/README.template.md

File: build.gradle [hunk 1]
```diff
@@ -34,7 +34,8 @@ def libraries = [
         "org.jetbrains.kotlinx:kotlinx-serialization-json-jvm",
         "org.jetbrains.kotlinx:kotlinx-serialization-cbor-jvm",
         "org.jetbrains.kotlinx:atomicfu-jvm",
-        "org.jetbrains.kotlinx:kotlinx-datetime-jvm"
+        "org.jetbrains.kotlinx:kotlinx-datetime-jvm",
+        "org.jetbrains.kotlinx:kotlinx-io-core-jvm"
```
Comment (modmuss50) [inline build.gradle:38]: This [appears](https://mvnrepository.com/artifact/org.jetbrains.kotlinx/kotlinx-io-core-jvm/0.5.1) to depend on `kotlinx-io-bytestring-jvm`

Comment (btwonion) [reply build.gradle]: should this also be listed in the readme?

Comment (modmuss50) [reply build.gradle]: Yes, I think so. All of the others are.

Comment (btwonion) [reply build.gradle]: resolved

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**